### PR TITLE
"don't delay" for PATH_RESPONSE

### LINKED ISSUE #4025: Suspect this needs to not delay?
(Broken out of #3214.)

> MUST respond
>   immediately by echoing the data/
- Of course it may not be able to immediately send anything:-) , but I suspect this needs to not delay?

### PR DESCRIPTION
**Pull request by martinthomson**
This is the maximal version of a clarification of this text.

Closes #4025.

### CODE REVIEW

**Comment by gorryfair** [comment]
WFM, and resolves the original NiT.

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -2174,8 +2174,10 @@ it can associate the peer's response with the corresponding PATH_CHALLENGE.
 
 ## Path Validation Responses
 
-On receiving a PATH_CHALLENGE frame, an endpoint MUST respond immediately by
-echoing the data contained in the PATH_CHALLENGE frame in a PATH_RESPONSE frame.
+On receiving a PATH_CHALLENGE frame, an endpoint MUST respond by echoing the
+data contained in the PATH_CHALLENGE frame in a PATH_RESPONSE frame.  An
+endpoint MUST NOT delay transmission of a packet containing a PATH_RESPONSE
+frame unless constrained by congestion control.
```
Comment (marten-seemann) [inline draft-ietf-quic-transport.md:2180]: Can you actually be constrained by congestion control on a new path?

Comment (ianswett) [reply draft-ietf-quic-transport.md:2180]: You can use PATH_CHALLENGE on existing paths as well, in which case congestion control could be a constraint.

Comment (martinthomson) [reply draft-ietf-quic-transport.md:2180]: It's true to say that any constraint is unlikely, but not impossible.

**Pull request merged**